ReSharper.FSharp/src/FSharp.Psi.Features/src/Formatter/FSharpReformatCode.fs

```diff
@@ -39,83 +36,44 @@ type FSharpReformatCode() =
-                      SpaceBeforeParameter = settings.SpaceBeforeParameter
-                      SpaceBeforeLowercaseInvocation = settings.SpaceBeforeLowercaseInvocation
-                      SpaceBeforeUppercaseInvocation = settings.SpaceBeforeUppercaseInvocation
-                      SpaceBeforeClassConstructor = settings.SpaceBeforeClassConstructor
-                      SpaceBeforeMember = settings.SpaceBeforeMember
-                      SpaceBeforeColon = settings.SpaceBeforeColon
-                      SpaceAfterComma = settings.SpaceAfterComma
-                      SpaceBeforeSemicolon = settings.SpaceBeforeSemicolon
-                      SpaceAfterSemicolon = settings.SpaceAfterSemicolon
-                      IndentOnTryWith = settings.IndentOnTryWith
-                      SpaceAroundDelimiter = settings.SpaceAroundDelimiter
-                      MaxIfThenElseShortWidth = settings.MaxIfThenElseShortWidth
-                      MaxInfixOperatorExpression = settings.MaxInfixOperatorExpression
-                      MaxRecordWidth = settings.MaxRecordWidth
-                      MaxArrayOrListWidth = settings.MaxArrayOrListWidth
-                      MaxValueBindingWidth = settings.MaxValueBindingWidth
-                      MaxFunctionBindingWidth = settings.MaxFunctionBindingWidth
-                      MultilineBlockBracketsOnSameColumn = settings.MultilineBlockBracketsOnSameColumn
-                      NewlineBetweenTypeDefinitionAndMembers = settings.NewlineBetweenTypeDefinitionAndMembers
-                      KeepIfThenInSameLine = settings.KeepIfThenInSameLine
-                      MaxElmishWidth = settings.MaxElmishWidth
-                      SingleArgumentWebMode = settings.SingleArgumentWebMode
-                      AlignFunctionSignatureToIndentation = settings.AlignFunctionSignatureToIndentation
-                      AlternativeLongMemberDefinitions = settings.AlternativeLongMemberDefinitions }
-
             let stamp = document.LastModificationStamp
             let modificationSide = TextModificationSide.NotSpecified
             let newLineText = sourceFile.DetectLineEnding().GetPresentation()
             let parsingOptions = checkerService.FcsProjectProvider.GetParsingOptions(sourceFile)
-            let checker = checkerService.Checker
 
             let change = 
                 if isNotNull rangeMarker then
                     try
                         let range = ofDocumentRange rangeMarker.DocumentRange
 
                         let formatted =
-                            CodeFormatter
-                                .FormatSelectionAsync(filePath, range, source, formatConfig, parsingOptions, checker)
-                                .RunAsTask()
+                            codeFormatterProvider
+                                .FormatSelection(filePath, range, text, settings, parsingOptions)
                                 .Replace("\r\n", newLineText)
```

Comment:
I wonder if it would be better to move this result processing the the other process too?